spring-integration-hazelcast/src/test/java/org/springframework/integration/hazelcast/inbound/HazelcastDistributedSQLInboundChannelAdapterTests.java

```diff
@@ -0,0 +1,174 @@
+
+	@Test
+	public void testDistributedSQLForOnlyENTRYIterationType() throws InterruptedException {
+		dsDistributedMap1.put(1, new User(1, "TestName1", "TestSurname1", 10));
+		dsDistributedMap1.put(2, new User(2, "TestName2", "TestSurname2", 20));
+		dsDistributedMap1.put(3, new User(3, "TestName3", "TestSurname3", 30));
+		dsDistributedMap1.put(4, new User(4, "TestName4", "TestSurname4", 40));
+		dsDistributedMap1.put(5, new User(5, "TestName5", "TestSurname5", 50));
+
+		Message<?> msg = null;
+		for (int i = 0; i < 10; i++) {
+			msg = dsMapChannel1.receive();
+			if (((Collection) msg.getPayload()).iterator().hasNext()) {
+				break;
+			}
+			Thread.sleep(1_000);
+		}
+
+		Assert.assertNotNull(msg);
+		Assert.assertNotNull(msg.getPayload());
+		Assert.assertTrue(msg.getPayload() instanceof Collection);
+		Assert.assertEquals(4, (((Map.Entry) ((Collection) msg.getPayload()).iterator()
+				.next()).getKey()));
+		Assert.assertEquals(4, ((User) ((Map.Entry) ((Collection) msg.getPayload())
+				.iterator().next()).getValue()).getId());
+		Assert.assertEquals("TestName4", ((User) ((Map.Entry) ((Collection) msg
+				.getPayload()).iterator().next()).getValue()).getName());
+		Assert.assertEquals("TestSurname4", ((User) ((Map.Entry) ((Collection) msg
+				.getPayload()).iterator().next()).getValue()).getSurname());
+	}
+
+	@Test
+	public void testDistributedSQLForOnlyKEYIterationType() throws InterruptedException {
+		dsDistributedMap2.put(1, new User(1, "TestName1", "TestSurname1", 10));
+		dsDistributedMap2.put(2, new User(2, "TestName2", "TestSurname2", 20));
+		dsDistributedMap2.put(3, new User(3, "TestName3", "TestSurname3", 30));
+
+		Message<?> msg = null;
+		for (int i = 0; i < 10; i++) {
+			msg = dsMapChannel2.receive();
+			if (((Collection) msg.getPayload()).iterator().hasNext()) {
+				break;
+			}
+			Thread.sleep(1_000);
+		}
+
+		Assert.assertNotNull(msg);
+		Assert.assertNotNull(msg.getPayload());
+		Assert.assertTrue(msg.getPayload() instanceof Collection);
+		Assert.assertEquals(1, ((Collection) msg.getPayload()).iterator().next());
+	}
+
+	@Test
+	public void testDistributedSQLForOnlyLOCAL_KEYIterationType()
+			throws InterruptedException {
+		dsDistributedMap3.put(1, new User(1, "TestName1", "TestSurname1", 10));
+		dsDistributedMap3.put(2, new User(2, "TestName2", "TestSurname2", 20));
+		dsDistributedMap3.put(3, new User(3, "TestName3", "TestSurname3", 30));
+
+		Message<?> msg = null;
+		for (int i = 0; i < 10; i++) {
+			msg = dsMapChannel3.receive();
+			if (((Collection) msg.getPayload()).iterator().hasNext()) {
+				break;
+			}
+			Thread.sleep(1_000);
+		}
+
+		Assert.assertNotNull(msg);
+		Assert.assertNotNull(msg.getPayload());
+		Assert.assertTrue(msg.getPayload() instanceof Collection);
+		Assert.assertEquals(2, ((Collection) msg.getPayload()).iterator().next());
+	}
+
+	@Test
+	public void testDistributedSQLForOnlyVALUEIterationType() throws InterruptedException {
```

Comment:
Duration to pass `4.002s`

Please, take a look to these tests to make them faster.

The whole metrics:
```
org.springframework.integration.hazelcast.inbound 	29 	0 	0 	7.319s 	100%
org.springframework.integration.hazelcast.outbound 	8 	0 	0 	0.047s 	100%
```